tensorboard/plugins/hparams/hparams_plugin.py

```diff
@@ -90,6 +102,107 @@ def is_active(self):
         return base_plugin.FrontendMetadata(element_name="tf-hparams-dashboard")
 
+    # ---- /download_data- -------------------------------------------------------
+    @wrappers.Request.application
+    def download_data_route(self, request):
+        response_format = request.args.get("format")
+        request_proto = _parse_request_argument(
+            request, api_pb2.ListSessionGroupsRequest
+        )
+        session_groups = list_session_groups.Handler(
+            self._context, request_proto
+        ).run()
+        experiment = get_experiment.Handler(self._context).run()
+        try:
+            body, mime_type = self.download_data_impl(
+                experiment, session_groups, response_format
+            )
+        except ValueError as e:
+            return http_util.Respond(
+                request=request,
+                content=str(e),
+                content_type="text/plain",
+                code=400,
+            )
+        return http_util.Respond(request, body, mime_type)
+
+    def download_data_impl(self, experiment, session_groups, response_format):
+        """Provides a response for downloading data for an HParams table.
+
+        Args:
+          experiment: Experiment proto.
+          session_groups: ListSessionGroupsResponse proto.
+          response_format: A string in the OutputFormat enum.
+
+        Raises:
+          ValueError: If the scalars plugin is not registered.
+
+        Returns:
+          2 entities:
+            - A JSON object response body.
+            - A mime type (string) for the response.
+        """
+        header = []
+        for hparam_info in experiment.hparam_infos:
+            header.append(hparam_info.display_name or hparam_info.name)
+
+        for metric_info in experiment.metric_infos:
+            header.append(metric_info.display_name or metric_info.name.tag)
+
+        rows = []
+
+        def _get_value(value):
+            if value.HasField("number_value"):
+                return value.number_value
+            if value.HasField("string_value"):
+                return value.string_value
+            if value.HasField("bool_value"):
+                return value.bool_value
+            return None
+
+        for group in session_groups.session_groups:
+            row = []
+            for hparam_info in experiment.hparam_infos:
+                row.append(_get_value(group.hparams[hparam_info.name]))
+            for metric_value in group.metric_values:
+                row.append(metric_value.value)
+            rows.append(row)
+
+        if response_format == OutputFormat.JSON:
+            mime_type = "application/json"
+            body = dict(header=header, rows=rows)
+        elif response_format == OutputFormat.LATEX:
+
+            def latex_format(value):
+                if isinstance(value, int):
+                    return str(value)
+                elif isinstance(value, float):
+                    return "%.3g" % value
```

Comment:
Numbers should be emitted in math mode, so something like `"$%.3g$"` is
a closer format string, but I’m not sure that that’s the exponent
formatting that we want. What do you think of something like

```python
import math
import re

_RE_EXPONENT = re.compile(r"e(?:\+|(-))0*([0-9]*)")


def latex_format(value):
    ...
    if isinstance(value, float):
        if math.isnan(value):
            return r"$\mathrm{NaN}$"
        if not math.isfinite(value):
            return r"$%s\infty$" % ("-" if value < 0 else "+")
        scientific = "%.3g" % value
        mathmode = _RE_EXPONENT.sub(r" \\cdot 10^{\1\2}", scientific)
        return "$%s$" % mathmode
```

or similar?

Sample output:

```tex
$1 \cdot 10^{-50}$
$1 \cdot 10^{-5}$
$0.01$
$0$
$100$
$1 \cdot 10^{5}$
$1 \cdot 10^{50}$
```